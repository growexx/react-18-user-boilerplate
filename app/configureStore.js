import { configureStore } from '@reduxjs/toolkit';
import { createBrowserHistory } from 'history';
import { createReduxHistoryContext } from 'redux-first-history';

import { repoApi } from './containers/HomePage/reposApiSlice';
import { appSlice } from './containers/App/slice';
import { apiSlice } from './apiSlice';
import languageSlice from './containers/LanguageProvider/slice';
import loginSlice from './containers/Auth/Login/slice';
import { authApi } from './containers/Auth/Login/authApiSlice';
import twoFactorAuthSlice from './containers/Auth/TwoFactorAuthentication/slice';

const { createReduxHistory, routerMiddleware, routerReducer } =
  createReduxHistoryContext({ history: createBrowserHistory() });

const store = configureStore({
  reducer: {
    router: routerReducer,
    app: appSlice,
    language: languageSlice,
    login: loginSlice,
    twoFactorAuth: twoFactorAuthSlice,
    repos: repoApi.reducer,
    auth: authApi.reducer,
    [apiSlice.reducerPath]: apiSlice.reducer,
  },
  middleware: (getDefaultMiddleware) => [
    ...getDefaultMiddleware(),
    apiSlice.middleware,
    routerMiddleware,
  ],
  devTools: process.env.NODE_ENV !== 'production',
});

const history = createReduxHistory(store);

export { store, history };
