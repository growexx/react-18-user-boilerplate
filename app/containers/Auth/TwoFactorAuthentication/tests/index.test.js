/**
 *
 * Tests for TwoFactorAuthentication
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { HistoryRouter as Router } from 'redux-first-history/rr6';
import history from 'utils/history';
import { store } from 'configureStore';
import { TwoFactorAuthentication } from '../index';
import Loadable from '../Loadable';
let globalStore;
const componentWrapper = Component =>
  render(
    <Provider store={globalStore}>
      <IntlProvider locale="en">
        <Router history={history}>
          <Component />
        </Router>
      </IntlProvider>
    </Provider>,
  );
describe('<TwoFactorAuthentication />', () => {
  beforeAll(() => {
    globalStore = store;
  });
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = componentWrapper(TwoFactorAuthentication);
    expect(firstChild).toMatchSnapshot();
  });
  it('Should render and match the snapshot Loadable', () => {
    const {
      container: { firstChild },
    } = componentWrapper(Loadable);
    expect(firstChild).toMatchSnapshot();
  });
});
