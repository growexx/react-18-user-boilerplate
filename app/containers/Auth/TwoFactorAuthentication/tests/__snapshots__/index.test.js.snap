// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TwoFactorAuthentication /> Should render and match the snapshot 1`] = `
.c1 {
  font-family: 'Montserrat',sans-serif;
  display: flex;
  flex: 1;
  flex-wrap: wrap;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

.c1 .twoFactorAuthenticationTitle {
  text-align: center;
  margin: 0;
  font-weight: 700;
  font-size: 42px;
  color: #4d186e;
}

.c0 {
  font-family: 'Montserrat',sans-serif;
  display: flex;
  flex-wrap: wrap;
}

.c0 .sideContainer {
  max-width: 513px;
  width: 100%;
  background: #4d186e;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 32px;
}

.c0 .sideContainer .registrationSideContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  color: #fff;
}

.c0 .sideContainer .registrationSideContainer .title {
  text-align: center;
  font-size: 42px;
  font-weight: 700;
  margin: 0;
}

.c0 .sideContainer .registrationSideContainer .subTitle {
  width: 60%;
  text-align: center;
  margin: 0;
}

.c0 .sideContainer .registrationSideContainer button {
  margin-top: 35px;
  height: 40px;
  width: 160px;
  background: transparent;
  border: 2px solid #fff;
  border-radius: 50px;
  text-align: center;
  color: #fff;
}

.c0 .sideContainer .registrationSideContainer button:hover {
  color: #fff;
  border-color: #fff;
}

<div>
  <div
    class="c0"
  >
    <div
      class="sideContainer"
    >
      <div
        class="loginContainer"
      >
        <div
          class="registrationSideContainer"
        >
          <p
            class="title"
          >
            New Here!
          </p>
          <p
            class="subTitle"
          >
            To join us, please sign up with your personal info
          </p>
          <a
            href="/register"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-byeoj0 ant-btn-default"
              type="button"
            >
              SIGN UP
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      class="c1"
    >
      <p
        class="twoFactorAuthenticationTitle"
      >
        Two Factor Authentication
      </p>
      <div
        style="display: flex; align-items: center;"
      >
        <input
          aria-label="Please enter OTP character 1"
          autocomplete="off"
          maxlength="1"
          style="width: 40px; text-align: center; height: 40px; border-radius: 15%; font-weight: bold; font-size: 18px;"
          type="text"
          value=""
        />
        <span>
          -
        </span>
        <input
          aria-label="Please enter OTP character 2"
          autocomplete="off"
          maxlength="1"
          style="width: 40px; text-align: center; height: 40px; border-radius: 15%; font-weight: bold; font-size: 18px;"
          type="text"
          value=""
        />
        <span>
          -
        </span>
        <input
          aria-label="Please enter OTP character 3"
          autocomplete="off"
          maxlength="1"
          style="width: 40px; text-align: center; height: 40px; border-radius: 15%; font-weight: bold; font-size: 18px;"
          type="text"
          value=""
        />
        <span>
          -
        </span>
        <input
          aria-label="Please enter OTP character 4"
          autocomplete="off"
          maxlength="1"
          style="width: 40px; text-align: center; height: 40px; border-radius: 15%; font-weight: bold; font-size: 18px;"
          type="text"
          value=""
        />
        <span>
          -
        </span>
        <input
          aria-label="Please enter OTP character 5"
          autocomplete="off"
          maxlength="1"
          style="width: 40px; text-align: center; height: 40px; border-radius: 15%; font-weight: bold; font-size: 18px;"
          type="text"
          value=""
        />
        <span>
          -
        </span>
        <input
          aria-label="Please enter OTP character 6"
          autocomplete="off"
          maxlength="1"
          style="width: 40px; text-align: center; height: 40px; border-radius: 15%; font-weight: bold; font-size: 18px;"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<TwoFactorAuthentication /> Should render and match the snapshot Loadable 1`] = `null`;
